safe_landing_planner/include/safe_landing_planner/waypoint_generator.hpp

```diff
@@ -0,0 +1,115 @@
+#pragma once
+
+#include <avoidance/usm.h>
+#include <safe_landing_planner/grid.hpp>
+
+#include <Eigen/Dense>
+
+#include <functional>
+#include <vector>
+
+namespace avoidance {
+
+const std::vector<Eigen::Vector2f> exploration_pattern = {
+    Eigen::Vector2f(1.f, 0.f),  Eigen::Vector2f(1.f, 1.f),
+    Eigen::Vector2f(0.f, 1.f),  Eigen::Vector2f(-1.f, 1.f),
+    Eigen::Vector2f(-1.f, 0.f), Eigen::Vector2f(-1.f, -1.f),
+    Eigen::Vector2f(0.f, -1.f), Eigen::Vector2f(1.f, -1.f)};
+
+enum class SLPState {
+  goTo,
+  loiter,
+  land,
+  altitudeChange,
+};
+std::string toString(SLPState state);  // for logging
+
+class WaypointGenerator : public usm::StateMachine<SLPState> {
+ public:
+  WaypointGenerator();
+  virtual ~WaypointGenerator() = default;
+
+  /**
+  * @brief     computes the setpoints to be sent to the FCU
+  **/
+  void calculateWaypoint();
+
+ protected:
+  // config
+  float yaw_setpoint_ = NAN;
+  float yaw_speed_setpoint_ = NAN;
+  float loiter_yaw_ = NAN;
+  float yaw_ = NAN;
+  float beta_ = 0.9f;
+  float landing_radius_ = 2.f;
+  float can_land_thr_ = 0.4f;
+  float loiter_height_ = 4.f;
+  float factor_exploration_ = 1.f;
+  float vertical_range_error_ = 1.f;
+  float spiral_width_ = 2.f;
+  float altitude_landing_area_percentile_ = -1.f;
+  int smoothing_land_cell_ = 2;
+
+  // state
+  bool trigger_reset_ = false;
+  SLPState prev_slp_state_ = SLPState::goTo;
+
+  bool is_land_waypoint_ = false;
+  bool decision_taken_ = false;
+  bool can_land_ = true;
+  bool update_smoothing_size_ = false;
+  bool explorarion_is_active_ = false;
+  int start_seq_landing_decision_ = 0;
+  int grid_slp_seq_ = 0;
+  int n_explored_pattern_ = -1;
+
+  Eigen::Vector3f position_ = Eigen::Vector3f(NAN, NAN, NAN);
+  Eigen::Vector3f goal_ = Eigen::Vector3f(NAN, NAN, NAN);
+  Eigen::Vector3f velocity_setpoint_ = Eigen::Vector3f(NAN, NAN, NAN);
+  Eigen::Vector3f loiter_position_ = Eigen::Vector3f(NAN, NAN, NAN);
+  Eigen::Vector3f exploration_anchor_ = Eigen::Vector3f(NAN, NAN, NAN);
+  Eigen::Vector3f vel_sp = Eigen::Vector3f(NAN, NAN, NAN);
```

Comment:
There's a `vel_sp` and a `velocity_setpoint_`... Smells like debug code that wasn't properly cleaned up

Reply:
@mrivi